### PR TITLE
rust: implement plugin based code generation

### PR DESCRIPTION
**Pull request by adsharma**
This allows us to keep complex use cases such as argument parsing
separate from the core transpiler.

Related to: https://github.com/adsharma/py2many/issues/179

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -40,10 +43,17 @@ jobs:
...
-      - name: Install tox
-        run: pip3 install tox
+      - name: Install Python packages
+        shell: bash
+        # Use argparse_dataclass from pypi once a new package has been uploaded
+        run: |
+          (pip3 install tox \
+           && git clone https://github.com/mivade/argparse_dataclass/ \
+           && (cd argparse_dataclass; python setup.py install) \
+           && rm -rf argparse_dataclass \
```
Comment (adsharma) [inline .github/workflows/main.yml:53]: @jayvdb - this isn't working as intended. I suppose tox still can't find `argparse_dataclass`.

Ideally upstream uploads a new package to pypi and we pick it up via `pip3 install`. Do you have any suggestions on the right way to do this?

Comment (jayvdb) [reply .github/workflows/main.yml:53]: in tox.ini, add `git+https://github.com/mivade/argparse_dataclass/` to the deps

**Comment by adsharma** [comment]
```
fatal error: 'vector' file not found
```

This looks like a new error that's unrelated to the diff in question.

**Comment by jayvdb** [comment]
I have a fix for `vector` and a bunch of related cpp include improvements.  I'll have it pushed up for review in approx 2 hrs.

**Pull request merged**